pywt/_extensions/_pywt.pyx

```diff
@@ -233,16 +241,30 @@ def families(int short=True):
     --------
     >>> import pywt
     >>> pywt.families()
-    ['haar', 'db', 'sym', 'coif', 'bior', 'rbio', 'dmey']
+    ['haar', 'db', 'sym', 'coif', 'bior', 'rbio', 'dmey', 'gaus', 'mexh', 'morl', 'cgau', 'shan', 'fbsp', 'cmor']
     >>> pywt.families(short=False)
-    ['Haar', 'Daubechies', 'Symlets', 'Coiflets', 'Biorthogonal', 'Reverse biorthogonal', 'Discrete Meyer (FIR Approximation)']
+    ['Haar', 'Daubechies', 'Symlets', 'Coiflets', 'Biorthogonal', 'Reverse biorthogonal', 'Discrete Meyer (FIR Approximation)', 'Gaussian', 'Mexican hat wavelet', 'Morlet wavelet', 'Complex Gaussian wavelets', 'Shannon wavelets', 'Frequency B-Spline wavelets', 'Complex Morlet wavelets']
 
     """
     if short:
         return __wfamily_list_short[:]
     return __wfamily_list_long[:]
 
 
+def DiscreteContinuousWavelet(name=u"", object filter_bank=None):
```

Comment:
@rgommers, @aaren 
I think this is the only other outstanding issue prior to merging this PR. @holgern has added a docstring since the original comment.

I agree that the naming suggested by @aaren is preferable, but the difficulty is that both `pywt` and likely a substantial amount of 3rd party code have statements like 

```python
if isinstance(wave, Wavelet):
```
that would be broken by switching the `Wavelet` class name to `DiscreteWavelet`. 

Unless there are strong objections to leaving the naming as is, I am inclined to merge as is.  The renaming and any potential deprecations, etc, can always be addressed in a separate future PR.